test/test.erl

```diff
@@ -6,16 +6,39 @@ should_pass_test_() ->
     %% user_types.erl references remote_types.erl
     %% it is not in the sourcemap of the DB so let's import it manually
     gradualizer_db:import_erl_files(["test/should_pass/user_types.erl"]),
-    run_tests_in("test/should_pass", ok).
+    map_erl_files(fun(File) ->
+            ?_assertEqual(ok, gradualizer:type_check_file(File))
+        end, "test/should_pass").
 
 should_fail_test_() ->
-    run_tests_in("test/should_fail", nok).
+    map_erl_files(fun(File) ->
+            ?_assertEqual(nok, gradualizer:type_check_file(File))
+        end, "test/should_fail").
 
-run_tests_in(Dir, ExpectedRes) ->
-    Files = filelib:wildcard(filename:join(Dir, "*.erl")),
+% Test succeeds if Gradualizer crashes or if it doesn't type check.
+% Doing so makes the test suite notify us whenever a known problem is resolved.
+known_problem_should_pass_test_() ->
+    map_erl_files(fun(File) ->
+        Result =
+            try gradualizer:type_check_file(File) of
+                V -> V
+            catch E -> nok
```

Comment:
this will only catch exception of type `throw` (doesn't type check) but not `error`(crash). So a crash of gradualizer would crash the unit test. (I would use `catch _:_ -> nok`) Or I might miss something.

Reply:
Aha, the point was to catch errors. If you have time please patch it, otherwise I'll do it tomorrow.